Today Extension/TodayExtensionCell.swift

```diff
@@ -0,0 +1,215 @@
+        }
+
+        if showLiveElements {
+            liveLabel.snp.makeConstraints { (make) in
+                make.bottom.equalToSuperview().inset(verticalMargin)
+                make.leading.equalTo(departureLabel)
+                make.height.equalTo(liveLabel.intrinsicContentSize.height)
+                make.width.equalTo(liveLabel.intrinsicContentSize.width)
+            }
+
+            liveIndicatorView.snp.makeConstraints { (make) in
+                make.centerY.equalTo(liveLabel.snp.centerY)
+                make.leading.equalTo(liveLabel.snp.trailing).offset(8)
+                make.trailing.lessThanOrEqualTo(rightMargin)
+                make.width.equalTo(liveIndicatorView.intrinsicContentSize.width)
+            }
+        }
+    }
+
+    func setUpCell(route: Route?, destination: String) {
+        destinationName = destination
+        if let route = route {
+            self.route = route
+            if let departDirection = (route.directions.filter { $0.type == .depart }).first {
+                busDirection = departDirection
+                busIcon = BusIcon(type: .directionSmall, number: departDirection.routeNumber)
+                contentView.addSubview(busIcon!)
+
+                setUpDepartureLabel()
+                setUpDestinationLabel()
+                setUpLiveElements()
+            } else { // there is no bus to this destination (i.e. only walking)
+                setUpNoRoute()
+            }
+        } else { // no route at all
+            setUpNoRoute()
+        }
+    }
+
+    func setUpNoRoute() {
+        showLiveElements = false
+        let noRouteLabel = UILabel()
+        noRouteLabel.font = .getFont(.regular, size: 14.0)
+        noRouteLabel.textColor = Colors.primaryText
+        noRouteLabel.textAlignment = .center
+        noRouteLabel.lineBreakMode = NSLineBreakMode.byTruncatingTail
+        noRouteLabel.text = "No routes available to \(destinationName)."
+
+        contentView.addSubview(noRouteLabel)
+
+        noRouteLabel.snp.makeConstraints { (make) in
+            make.centerY.equalToSuperview()
+            make.centerX.equalToSuperview()
+            make.leading.equalToSuperview().inset(12.0)
+            make.trailing.equalToSuperview().inset(12.0)
+        }
+    }
+
+    func setUpDepartureLabel() {
+        departureLabel.lineBreakMode = NSLineBreakMode.byTruncatingTail
+        departureLabel.font = .getFont(.medium, size: 16.0)
+        departureLabel.textColor = Colors.primaryText
+        departureLabel.font = .getFont(.medium, size: 16.0)
+        departureLabel.textColor = Colors.primaryText
+        if let route = route {
+            departureLabel.text = "To \(route.endName)"
+        }
+    }
+
+    private func getDelayState(fromDirection direction: Direction) -> DelayState {
+        let departTime = direction.startTime
+        if let delay = direction.delay {
+            let delayedDepartTime = departTime.addingTimeInterval(TimeInterval(delay))
```

Comment:
Are we absolutely sure that delay could never be negative (e.g. bus is running early?) Might want to use logic to specifically check that that it's late if delatedDepartTime (departTime with delay) is later than departTime.

Reply:
If the bus is running early, how do we want the cell to look? Just change the departure time to match the early time and report the bus as on time?

Reply:
Yeah!